### DISCUSSION TITLE
open applications automatically

### DISCUSSION THREAD

**Discussion by RiteshChepuri** [i3/i3#5545] [Q&A] [answered]
how to open specific applications on specific workspaces for every time i log into i3 wm . So far i tried this its opening all the applications in workspace 3.Need help!!!
`
exec --no-startup-id i3-msg 'workspace $ws1; exec alacritty'
exec --no-startup-id i3-msg 'workspace $ws2; exec brave'
exec --no-startup-id i3-msg 'workspace $ws3; exec firefox'

`

**Comment by otakutyrant** [comment]
`exec --no-startup-id` is enough to autostart some applications. You just need to assign some applications to specific workspaces in config, see [assign_workspace](https://i3wm.org/docs/userguide.html#assign_workspace).

**Reply by RiteshChepuri** [reply]
Just `exec_always alacritty` works? Where do we need to specify the workspace numbers?

**Reply by otakutyrant** [reply]
I do not know what `exec_always` is.

`$ws1` is an alias of the first workspace because the default config contains `set $ws1 "1"`. You can assign applicantions to either if your config contains `set $ws1 "1"`. You could read the default config from begining to end.

**Reply by okraits** [reply, marked as answer]
Please read the userguide: 

https://i3wm.org/docs/userguide.html
https://i3wm.org/docs/userguide.html#_automatically_starting_applications_on_i3_startup
https://i3wm.org/docs/userguide.html#assign_workspace

It will answer almost all questions you'll have about i3.
